File: 241219/그 요일은/the-day-of-the-day.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt(), d1 = sc.nextInt(), m2 = sc.nextInt(), d2 = sc.nextInt();
        String A = sc.next();
        
        int day1 = getDays(m1, d1);
        int day2 = getDays(m2, d2);
        int ifMon = A.equals("Mon")? 1:0;

        System.out.println((day2-day1)/7 + ifMon);
    }

    private static int getDays(int m, int d){
        int days = 0;
        int[] lastDay = new int[]{31,29,31,30,31,30,31,31,30,31,30,31};
        for(int i=0; i<m; i++){
            days += lastDay[i];
        }
        return days + d;
    }

    private static Map<String, Integer> yoil = new HashMap<String, Integer>(){{
            put("Mon", 1);
            put("Tue", 2);
            put("Wed", 3);
            put("Thu", 4);
            put("Fri", 5);
            put("Sat", 6);
            put("Sun", 7);
        }};
}

File: 241219/그 요일은/the-day-of-the-day.py
m1, d1, m2, d2 = map(int, input().split())
day = input()
days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
days_dict = {days:idx for idx, days in enumerate(days)}
idx = days_dict[day]

def calcDate(m, d):
    sum2 = 0
    for i in range(1, m):
        if i >= 8:
            if i % 2 == 0:
                sum2 += 31
            else:
                sum2 += 30
        else:
            if i % 2 == 0:
                if i == 2:
                    sum2 += 29
                else:
                    sum2 += 30
            else:
                sum2 += 31
    sum2 += d
    return sum2

sum1 = calcDate(m1, d1)
sum2 = calcDate(m2, d2)

sum0 = sum2-sum1

count = sum0//7
if sum0%7 >= idx:
    count += 1

print(count)